src/hotspot/cpu/s390/vmreg_s390.inline.hpp

```diff
@@ -26,18 +26,15 @@
 #ifndef CPU_S390_VMREG_S390_INLINE_HPP
 #define CPU_S390_VMREG_S390_INLINE_HPP
 
-inline VMReg RegisterImpl::as_VMReg() {
-  if (this == noreg) {
-    return VMRegImpl::Bad();
-  }
```

Comment:
Although @TheRealMDoerr  you've reviewed the changes. Do you think this getting rid from this check is okay? No other arch have this & this was an Typecasting error for us, after these changes.

Reply:
Fine with me. PPC64 still has it, but if it's not used, I think it's ok to remove it. We have the `assert(is_valid(), "invalid register")` in `encoding()` to catch it.
If you prefer to keep it, you need to use `*this`.